jib-core/src/test/java/com/google/cloud/tools/jib/tar/TarStreamBuilderTest.java

```diff
@@ -66,37 +108,57 @@ public void setUp() throws IOException, URISyntaxException {
             "some/really/long/path/that/exceeds/100/characters/abcdefghijklmnopqrstuvwxyz0123456789012345678901234567890"));
   }
 
-  @Test
-  public void testToBlob() throws IOException {
+  /** Creates a TarStreamBuilder using Strings. */
+  private void setUpWithStrings() {
+    // Prepares a test TarStreamBuilder.
+    testTarStreamBuilder.addEntry(fileAContents, "some/path/to/resourceFileA");
+    testTarStreamBuilder.addEntry(fileBContents, "crepecake");
+    testTarStreamBuilder.addEntry(new TarArchiveEntry(directoryA.toFile(), "some/path/to"));
+    testTarStreamBuilder.addEntry(
+        fileAContents,
+        "some/really/long/path/that/exceeds/100/characters/abcdefghijklmnopqrstuvwxyz0123456789012345678901234567890");
+  }
+
+  /** Creates a TarStreamBuilder using Strings and TarArchiveEntries. */
+  private void setUpWithStringsAndTarEntries() {
+    // Prepares a test TarStreamBuilder.
+    testTarStreamBuilder.addEntry(fileAContents, "some/path/to/resourceFileA");
+    testTarStreamBuilder.addEntry(new TarArchiveEntry(fileB.toFile(), "crepecake"));
+    testTarStreamBuilder.addEntry(new TarArchiveEntry(directoryA.toFile(), "some/path/to"));
+    testTarStreamBuilder.addEntry(
+        fileAContents,
+        "some/really/long/path/that/exceeds/100/characters/abcdefghijklmnopqrstuvwxyz0123456789012345678901234567890");
+  }
+
+  /** Creates a compressed blob from the TarStreamBuilder and verifies it. */
+  private void verifyBlobWithCompression() throws IOException {
```

Comment:
This is fine since it's just a test helper, but since theres only two line difference between these functions, it could just be one function with a parameter "compress". Then, like line 140 could be `blob.writeTo(compress?new GZIPOutputStream(tarByteOutputStream):tarByteOutputStream)`.